The SYN-C Brutus, a Derelict Syndicate Corvette

massaheartsu opened the pull request:
Idly drifting, red lights on her outer hull flickering, turrets still live.
## About The Pull Request

![image](https://user-images.githubusercontent.com/128642031/227725779-26ed033d-c3cc-4385-876c-2f2ca6868689.png)

This PR adds the ruin "infested_frigate.dmm", and the changes necessary. Originally, I had made a ruin for Void Crew out of a VC/Shiptest ship called the SYN-C Blackbeard. I decided to start from scratch, beef it up, and tell tell an authentic SS13 story between the vending machines flickering on in dark corners.

I spent a lot of time going over the balancing details as far as accessing the ship and its spaces goes, adjusting values, access, factions, power of the loot, and loot drops. I feel I've actually been rather conservative, and that I could stand to put at least a more powerful, fun item in there, but there is cool stuff. 

This was my first attempt at something like this, so I expect to correct a bunch of mistakes!

## Why It's Good For The Game

Map screenshot:

![image](https://user-images.githubusercontent.com/128642031/227761138-4d5443d9-4548-4a8c-a88a-951a3c8cdce6.png)

We needed more scifi ruins, and more interesting ruins in general, and here we are! This is a dangerous ruin that's packed densely with a variety of random spawning loot that's could be quite useful! The wreckage and blood tell the story of a crew that failed to escape a very ugly place. 

I went for something familiar, Xenomorph extermination. I also made sure that the Xenomorph corpses can't be used to force a Xenomorph outbreak. The queen gibs on death so no egg sac, as does the drone, and she has very special drops when she's taken down! Similar concerns were taken with much of the ruin. 

![image](https://user-images.githubusercontent.com/128642031/227761508-794ca833-1443-4f63-9957-64829ba80db1.png)

Volunteer for a tour aboard the Brutus!

### Mapping March

Ckey to receive rewards: Massa100

## Changelog

:cl:
add: Added a new ruin!
/:cl:

Feedback, pointers, advice is extremely welcome!

massaheartsu committed:
Infested Syndicate Frigate Ruin

This is the rough draft to a ruin I made.

It's based off of a ship from shiptest, with many substantial changes, that I then destroyed and filled with dead dudes and monsters and relatively interesting trash.

massaheartsu committed:
Update infested_frigate.dmm

massaheartsu committed:
Update infested_frigate.dmm

massaheartsu committed:
Update infested_frigate.dmm

massaheartsu committed:
Update infested_frigate.dmm

massaheartsu committed:
Update infested_frigate.dmm

major finalizations of various kinds, tiny attention to details, shit i did not keep track of lmao

massaheartsu committed:
Update infested_frigate.dmm

massaheartsu committed:
final commits before pr

final balance changes, many can be listed later

did as much debugging as i could, no more polishing a turd

massaheartsu committed:
oh my god final rough draft

well it's time to pr now i made it work i think

massaheartsu committed:
Merge branch 'master' into branched-infested-frigate

Cheshify commented:
As for the actual area of the shuttle, have it match the walls, you don't want it going outside.

nukularpower commented:
love to see randomized loot for sure!  sweet

Addust commented:
nice, a xeno-infested blackbeard. would love to see more Voidcrew presence inside the space of /tg/ and at the same time its interior design is a unique spin on the regular SYN-C Blackbeard.

Fikou commented on code/modules/mapfluff/ruins/spaceruin_code/infested_frigate.dm:
```diff
@@ -0,0 +1,2 @@
+/area/ruin/space/infested_frigate
+    name = "SYN-C Brutus"
```
you can put this in `code/game/area/areas/ruins/space.dm`

massaheartsu replied:
if i did this do i hit resolve conversation

Fikou replied:
yes

Fikou replied:
but you need to delete it from here dont add it twice
and then delete this file

Fikou reviewed (changes requested):
needs a map_template

san7890 commented:
post a screenshot without areas? thank you

massaheartsu committed:
Commits to make it real!

I:

- Adjusted a few final things for balance
- Edited my PR for presentation's sake and included some extra info
- added a map_template
- did what fikou said to do

massaheartsu committed:
Merge branch 'master' into branched-infested-frigate

massaheartsu commented:
I did the changes I could, made sure it all worked?! Did what the comments here said, booted up several times until it spawned naturally in space.

YuliceTheDruid commented:
I Am Looking Respectfully.

massaheartsu committed:
Update infested_frigate.dmm

final debugging to get rid of active turfs

massaheartsu committed:
Merge branch 'branched-infested-frigate' of https://github.com/massaheartsu/massastation into branched-infested-frigate

massaheartsu commented:
> I Am Looking Respectfully.

thank you! i fixed AT on round start and made sure they worked with this last commit

Singul0 commented on code/game/area/areas/ruins/space.dm:
```diff
@@ -578,3 +578,6 @@
 	name = "\improper Traveler's Rest"
+
+/area/ruin/space/infested_frigate
+    name = "SYN-C Brutus"
```
```suggestion
 name = "SYN-C Brutus"
```

massaheartsu committed:
fixed linters

hopefully

Singul0 commented on code/modules/mapfluff/ruins/spaceruin_code/infested_frigate.dm:
Don't forget to delete this file after you moved it to  `code/game/area/areas/ruins/space.dm` (unless you want to add holodisk/paper fluff further down the road). Remember to copy over the comment for it too.

Likteer commented:
> I also made sure that the Xenomorph corpses can't be used to force a Xenomorph outbreak. The queen gibs on death so no egg sac,

I am positive simplemob benos were already neutered (lavaland, charlie station). They are slow, attack in melee or fire acid a a plain dmg attack and yield no organs.

massaheartsu committed:
deleted the mapfluff

massaheartsu commented:
I did the last changes I could, deleted the mapfluff too. Thanks frens.

Watermelon914 commented on _maps/RandomRuins/SpaceRuins/infested_frigate.dmm:
```diff
@@ -0,0 +1,5162 @@
+	name = "crew supplies"
+	},
+/obj/effect/spawner/random/aimodule/harmful,
+/obj/effect/spawner/random/contraband,
+/obj/effect/spawner/random/contraband,
+/obj/effect/spawner/random/contraband,
+/obj/effect/spawner/random/contraband,
```
I notice you have these spawners inside the map, but /obj/effect/spawner/random/contraband does not spawn anything

Watermelon914 commented on _maps/RandomRuins/SpaceRuins/infested_frigate.dmm:
```diff
@@ -0,0 +1,5162 @@
+/obj/item/clothing/mask/facehugger/impregnated,
+/obj/machinery/power/apc/auto_name/directional/east,
+/obj/structure/cable,
+/obj/effect/spawner/random/exotic/antag_gear,
```
I don't think you should be spawning in antag gear

Fikou replied:
its not actual antag gear. this is fine

massaheartsu replied:
I actually looked at the loot tables of these randomized items and they're what made me feel I could make this place zestier. The actual rolls here are pretty minor.

Watermelon914 commented on _maps/RandomRuins/SpaceRuins/infested_frigate.dmm:
```diff
@@ -0,0 +1,5162 @@
+	pixel_x = 28
+	},
+/obj/item/storage/toolbox/syndicate,
+/obj/effect/spawner/random/exotic/tool,
```
Don't really like the idea of spawning in alien tools here, especially since it's not an alien ship.

Watermelon914 commented on _maps/RandomRuins/SpaceRuins/infested_frigate.dmm:
```diff
@@ -0,0 +1,5162 @@
+/obj/machinery/atmospherics/components/unary/vent_pump/on{
+	dir = 1
+	},
+/obj/effect/spawner/random/contraband/armory{
```
I definitely think you should not be spawning in weapons, remove this please

massaheartsu replied:
It's inside a sealed and heavily guarded armory on the inside of the heavily guarded ship though!

massaheartsu commented:
> needs a map_template



> > I also made sure that the Xenomorph corpses can't be used to force a Xenomorph outbreak. The queen gibs on death so no egg sac,
> 
> I am positive simplemob benos were already neutered (lavaland, charlie station). They are slow, attack in melee or fire acid a a plain dmg attack and yield no organs.

They do, I changed them all back 2 days ago. The queen still gibs but she's just a little different.

massaheartsu committed:
removed a specific spawner i didn't feel i had to, removed useless spawners, removed out of place items

massaheartsu committed:
Merge branch 'master' into branched-infested-frigate

massaheartsu committed:
Merge branch 'master' into branched-infested-frigate

massaheartsu committed:
Merge branch 'master' into branched-infested-frigate

massaheartsu committed:
Merge branch 'master' into branched-infested-frigate

massaheartsu committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into branched-infested-frigate

massaheartsu committed:
Merge branch 'master' into branched-infested-frigate

massaheartsu committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into branched-infested-frigate

massaheartsu committed:
fixed merge conflicts

once again

massaheartsu committed:
Merge branch 'master' into branched-infested-frigate

massaheartsu committed:
Merge branch 'master' into branched-infested-frigate

massaheartsu committed:
fixed small error

dunno what the compile map error was

massaheartsu committed:
fixed compile map error

removed a fucking hat apparently

massaheartsu committed:
fixed one last thing

i actually fixed the space indentation now

massaheartsu committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into branched-infested-frigate

massaheartsu committed:
Merge branch 'master' into branched-infested-frigate

massaheartsu committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into branched-infested-frigate

Maurukas commented on code/game/area/areas/ruins/space.dm:
```diff
@@ -648,3 +648,7 @@
 	name = "\improper outlet cult corruption"
+
+//SYN-C Brutus, derelict frigate
+/area/ruin/space/infested_frigate
```
```suggestion
/area/ruin/space/has_grav/infested_frigate
```
You'll also need to update the map

massaheartsu replied:
Done!

Maurukas commented on _maps/RandomRuins/SpaceRuins/infested_frigate.dmm:
```diff
@@ -0,0 +1,5149 @@
+	dir = 1
+	},
+/obj/machinery/atmospherics/pipe/smart/manifold4w/supply/hidden,
+/obj/machinery/door/airlock/highsecurity{
```
For welded doors there is a helper: `/obj/effect/mapping_helpers/airlock/welded` which you can drop on top of the door to weld it. Please use this. There are a couple of these on the map.

There is also an update paths that will do this for you in the `tools/UpdatePaths/Scripts. Find '74206_airlock_weld_help' and drag it onto the UpdatePaths batch file to run it.

massaheartsu replied:
Done!

Maurukas reviewed (changes requested):
Have you done this before? Because this is really good. Creative use of a lot of the mapping tools, lots of detailed varedits and little details. Actually rather challengiing, and with a pretty good loot balance of mostly cosmetic goodies and a few low-threat items.

There's a couple minor technical issues - you've got a varedited area, which is the one thing you can't edit in a map file. I've put a suggestion for how to fix that in the areas file, but you'll also need to replace the area in map. You've also used varedits instead of helpers in a couple places, and we generally want to use helpers when they're available - they make it clearer to future mappers what's being done on a tile.

massaheartsu committed:
Update code/game/area/areas/ruins/space.dm

Co-authored-by: Vire <66576896+Maurukas@users.noreply.github.com>

massaheartsu committed:
Fixed maptainer requests

Fixed varedited area, ran a bunch of times to make sure it spawns naturally and works well, replaced welded varedit doors with helpers.

massaheartsu committed:
Merge branch 'master' into branched-infested-frigate

massaheartsu committed:
Merge branch 'master' into branched-infested-frigate

massaheartsu committed:
Merge branch 'master' into branched-infested-frigate

massaheartsu committed:
fixed a linter error

forgot to replace spawner structures on 4 windows, i checked the other windows

massaheartsu committed:
Update infested_frigate.dmm

fixed this shit again
